shared/dev/dumb-sheet/render.native.js

```diff
@@ -131,26 +132,32 @@ class DumbSheetRender extends Component<void, Props, any> {
     }
     const {component, mock} = this._getComponent(null, this.state.testIndex)
 
-    this._updateMockStore(mock.mockStore)
+    if (!this._updateMockStore(mock.mockStore)) return null
+
     console.log('test render idx', this.state.testIndex, component)
     return <Box>{this._makeStoreWrapper(component)}</Box>
   }
 
   _updateMockStore(mockStore: any) {
-    if (mockStore) {
-      if (!this._mockStore) {
-        this._mockStore = createStore(old => mockStore, mockStore)
+    if (mockStore === this._mockStore) return true
+
+    setImmediate(() => {
```

Comment:
calling replaceReducer will actually fire off connectors which causes react to be very unhappy if we're mounting / unmounting. instead we call setState and manage this slightly more